### PR TITLE
request an early login on explicit upload [ready for review]

### PR DESCRIPTION
**Pull request by matkoniecz**
fixes #1446 - upload button was not working for unlogged users with low star count
affected people not logged in, with low star count and disabled autosync

-----

My work on this pull request was sponsored by a NGI Zero Discovery [grant](https://www.openstreetmap.org/user/Mateusz%20Konieczny/diary/368849)

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/statistics/AnswersCounter.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 	private View unsyncedContainer;
 
 	private boolean isFirstUpdateDone;
-	private boolean isAutosync;
+	public boolean isAutosync;
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/statistics/AnswersCounter.java]: Instead please put `isAutosync` from `onStart()` in `MainActivity` into own function and use that.

```java
private Prefs.Autosync isAutosync()
{
  return Prefs.Autosync.valueOf(prefs.getString(Prefs.AUTOSYNC,"ON")) == Prefs.Autosync.ON
}
```

**Comment by matkoniecz** [comment]
>  because there is `Prefs.Autosync.WIFI`

Argh, I even remembered this time about it and tested, but only to ensure that manual upload still works.

**Comment by matkoniecz** [comment]
Updated again and I again think that it is correct (due to redoing everything in much simpler I went with `reset --hard`, but I may submit version with full history if that is preferable)

**Comment by westnordost** [comment]
Ok, looks good!

**Comment by matkoniecz** [comment]
Thanks for a re-re-re-review! I am so happy that it is now correct (hopefully it is really correct).

**Comment by westnordost** [comment]
maybe you should test it ;-) (I didn't, got a big heap of uncommitted changes in a spiraling-out-of-control refactor of the whole upload logic)

**Comment by matkoniecz** [comment]
I tested it. Unfortunately earlier attempts were also tested, so it is hard to convince myself that this time everything was caught...

> refactor of the whole upload logic

Sounds scary, especially if that is together with way splitting.

**Pull request merged**